carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/deployer/MSSDeployer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.mss.internal.deployer;
+
+
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.deployment.Artifact;
+import org.wso2.carbon.deployment.ArtifactType;
+import org.wso2.carbon.deployment.exception.CarbonDeploymentException;
+import org.wso2.carbon.deployment.spi.Deployer;
+import org.wso2.carbon.mss.internal.MicroservicesRegistry;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Implementation of the MSS service deployer. This will be picked by the
+ * DeploymentEngine service according to the whiteboard pattern to deploy
+ * MSS POJO artifacts
+ */
+@Component(
+        name = "org.wso2.carbon.mss.internal.deployer.MSSDeployer",
+        service = Deployer.class,
+        immediate = true
+)
+@SuppressWarnings("unused")
+public class MSSDeployer implements Deployer {
+    private static final Logger log = LoggerFactory.getLogger(MSSDeployer.class);
+    private static final String DEPLOYMENT_PATH = "file:mss";
+    private URL deploymentLocation;
+    private ArtifactType artifactType;
+    private HashMap<Object, List<Object>> deployedArtifacts = new HashMap<>();
+
+    public void init() {
+        log.info("MSSDeployer initializing");
+        artifactType = new ArtifactType<>("MSS");
+        try {
+            deploymentLocation = new URL(DEPLOYMENT_PATH);
+        } catch (MalformedURLException e) {
+            log.error("MSS deployer location error");
+        }
+    }
+
+    /**
+     * Deploy and artifact in the netty-http service
```

Comment:
Samiyuru, please make the relevant changes related to the comments people
are making & send PRs.

Thanks
Azeez

On Sun, Jan 10, 2016 at 11:26 AM, Anjana Fernando <notifications@github.com>
wrote:

> In
> carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/deployer/MSSDeployer.java
> <https://github.com/wso2/product-mss/pull/3#discussion_r49271958>:
>
> > +    private URL deploymentLocation;
> > +    private ArtifactType artifactType;
> > +    private HashMap<Object, List<Object>> deployedArtifacts = new HashMap<>();
> > +
> > +    public void init() {
> > +        log.info("MSSDeployer initializing");
> > +        artifactType = new ArtifactType<>("MSS");
> > +        try {
> > +            deploymentLocation = new URL(DEPLOYMENT_PATH);
> > +        } catch (MalformedURLException e) {
> > +            log.error("MSS deployer location error");
> > +        }
> > +    }
> > +
> > +    /**
> > +     * Deploy and artifact in the netty-http service
>
> Put the Javadoc comments in the correct way, and be consistent. For
> example, the first line should end with a full stop. The lines needs to
> start with a capital letter. If a method has a Javadoc comment, put the
> full one, without some being explaining just a parameter. Fix in all places.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/product-mss/pull/3/files#r49271958>.
>



-- 
*Afkham Azeez*
Director of Architecture; WSO2, Inc.; http://wso2.com
Member; Apache Software Foundation; http://www.apache.org/
* <http://www.apache.org/>*
*email: **azeez@wso2.com* <azeez@wso2.com>
* cell: +94 77 3320919blog: **http://blog.afkham.org*
<http://blog.afkham.org>
*twitter: **http://twitter.com/afkham_azeez*
<http://twitter.com/afkham_azeez>
*linked-in: **http://lk.linkedin.com/in/afkhamazeez
<http://lk.linkedin.com/in/afkhamazeez>*

*Lean . Enterprise . Middleware*